src/commands/moderation/ban.ts

```diff
@@ -1,120 +1,154 @@
+import { Colors, TextChannel, User } from 'discord.js';
+import moment from 'moment';
 import { CommandDefinition } from '../../lib/command';
 import { Channels, CommandCategory } from '../../constants';
-import { makeEmbed } from '../../lib/embed';
+import { makeEmbed, makeLines } from '../../lib/embed';
 
-type UserLike = User | Snowflake
+const modLogEmbed = (formattedDate, moderator: User, user: User, reason: string) => makeEmbed({
+    color: Colors.Red,
+    author: {
+        name: `[BANNED] ${user.tag}`,
+        iconURL: user.displayAvatarURL(),
+    },
+    fields: [
+        {
+            name: 'Member',
+            value: user.toString(),
+        },
+        {
+            name: 'Moderator',
+            value: moderator.toString(),
+        },
+        {
+            name: 'Reason',
+            value: `\u200B${reason}`,
+        },
+        {
+            inline: false,
+            name: 'Date',
+            value: formattedDate,
+        },
+    ],
+    footer: { text: ` User ID: ${user.id}` },
+});
+
+const successfulBanEmbed = (user: User, reason: string) => makeEmbed({
+    title: 'User Successfully Banned',
+    color: Colors.Green,
+    fields: [
+        {
+            inline: true,
+            name: 'Username',
+            value: user.toString(),
+        },
+        {
+            inline: true,
+            name: 'ID',
+            value: (user instanceof User) ? user.id : user,
+        },
+        {
+            inline: false,
+            name: 'Reason',
+            value: reason,
+        },
+    ],
+});
+
+const failedBanEmbed = (user: User, error: any) => makeEmbed({
+    title: 'Failed to Ban User',
+    color: Colors.Red,
+    fields: [
+        {
+            inline: true,
+            name: 'Username',
+            value: user.toString(),
+        },
+        {
+            inline: true,
+            name: 'Username',
+            value: user.toString(),
+        },
+        {
+            inline: true,
+            name: 'ID',
+            value: user.id,
+        },
+        {
+            inline: false,
+            name: 'Error',
+            value: `\u200B${error}`,
+        },
+    ],
+});
+
+const dmEmbed = (formattedDate, moderator: User, reason: string) => makeEmbed({
+    title: 'You have been banned from FlyByWire Simulations',
+    fields: [
+        {
+            inline: false,
+            name: 'Moderator',
+            value: moderator.toString(),
+        },
+        {
+            inline: false,
+            name: 'Reason',
+            value: reason,
+        },
+        {
+            inline: false,
+            name: 'Date',
+            value: formattedDate,
+        },
+        {
+            inline: false,
+            name: 'Appeal',
+            //value: `If you would like to appeal your ban, please fill out [this form.](${process.env.BAN_APPEAL_URL})`,
+            value: 'If you would like to appeal your ban, please fill out [this form.](https://www.google.co.uk/)',
```

Comment:
Is this actually an official form or could you just say to appeal it contact the moderator who banned?

Reply:
An actual form, URL will be stored with github secrets. There is also a webhook integration in a channel which drops a embed when a form is submitted